File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/main/command/argument/CommandArgPlayer.java
package pize.tests.voxelgame.main.command.argument;

import pize.tests.voxelgame.main.net.PlayerProfile;
import pize.tests.voxelgame.server.Server;
import pize.tests.voxelgame.main.command.source.CommandSource;
import pize.tests.voxelgame.server.player.ServerPlayer;

public class CommandArgPlayer extends CommandArg{
    
    // Результат парсинга
    private ServerPlayer player;
    
    @Override
    public int parse(String remainingChars, CommandSource source, Server server){
        // Разделяем оставшуюся часть команды на части
        final String[] args = remainingChars.split(" ");
        
        // Если количество частей меньше 1 (имя игрока), завершить парсинг
        if(args.length < 1)
            return 0;
        
        // Находим игрока
        final String playerName = args[0];
        if(PlayerProfile.isNameInvalid(playerName))
            return 0;
        
        player = server.getPlayerList().getPlayer(playerName);
        if(player == null)
            return 0;
        
        return playerName.length();
    }
    
    public ServerPlayer getPlayer(){
        return player;
    }
    
}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/main/command/argument/CommandArgPlayer.kt
package pize.tests.voxelgame.main.command.argument

import pize.tests.voxelgame.main.command.source.CommandSource
import pize.tests.voxelgame.main.net.PlayerProfile
import pize.tests.voxelgame.server.Server
import pize.tests.voxelgame.server.player.ServerPlayer

class CommandArgPlayer : CommandArg() {
    // Результат парсинга
    var player: ServerPlayer? = null
        private set

    override fun parse(remainingChars: String, source: CommandSource, server: Server): Int {
        // Разделяем оставшуюся часть команды на части
        val args = remainingChars.split(" ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()

        // Если количество частей меньше 1 (имя игрока), завершить парсинг
        if (args.size < 1) return 0

        // Находим игрока
        val playerName = args[0]
        if (PlayerProfile.Companion.isNameInvalid(playerName)) return 0
        player = server.playerList.getPlayer(playerName)
        return if (player == null) 0 else playerName.length
    }
}
